content/en/docs/tasks/administer-cluster/pids.md

```diff
@@ -0,0 +1,44 @@
+---
+reviewers:
+- derekwaynecarr
+- dashpole
+- RobertKrawitz
+title: Pid Limiting
+content_template: templates/concept
+---
+
+{{% capture overview %}}
+{{< feature-state state="beta" >}}
+
+This page explains how to configure pid limiting with the `kubelet`.
+
+Pids are a fundamental resource on Linux hosts. It is trivial to hit the task
```

Comment:
Should we suggest ways to make more process IDs available? 

Eg, on Linux  set `kernel.pid_max` via `sysctl`, modern kernels support 2<sup>22</sup>-1 process IDs.

Reply:
Simply raising the process limit isn't necessarily the right answer either -- it probably means more time spent doing accounting, scanning the process table, etc.  Not to mention the resources that are consumed by that many processes.

Before we go into that, I think we want to decide what level of discussion we want here.

Reply:
The aim I'd have here is to let a less experienced administrator realize that:
- they can have kubelet & their container runtime limit how many pids are available to a pod
- they can raise the system-wide pid ceiling from the default, which is typically quite low

and that making both changes together can work well.
(A high pid ceiling, system wide, helps avoid collisions when IDs are reused, and a low limit per pod protects the other pods and the rest of the system).

Reply:
The system-wide pid-max is typically 32K ([bigger on machines with more than 32 CPU threads available](https://unix.stackexchange.com/questions/500255/i-see-pids-above-400-000-why-is-that-does-it-indicate-something-is-wrong)).  Whether that's too small on any given system depends, but I'm not convinced it would be a good idea for an inexperienced admin to simply raise the process limit without an understanding of the workload being run and analyzing the entire system's capacity.

The per-pod limit only protects the system to the extent that the number of pods is limited.

The node limit has nothing to do with collisions; it's simply a hard upper limit on the number of simultaneous tasks in existence on a node.